src/utils.ts

```diff
@@ -862,34 +862,34 @@ export function makeInterruptibleAsyncInterval(
 
   function wake() {
     const currentTime = clock();
-    const nextScheduledCallTime = lastCallTime + interval;
-    const timeUntilNextCall = nextScheduledCallTime - currentTime;
+    const timeSinceLastWake = currentTime - lastWakeTime;
+    const timeSinceLastCall = currentTime - lastCallTime;
+    const timeUntilNextCall = interval - timeSinceLastCall;
```

Comment:
why did the logic for `timeUntilNextCall` need a refactor? it's equivalent to the logic that was already in place, just using `nextScheduledCallTime` instead of `timeSinceLastCall` as far as I can tell, i.e., the previous logic had `(lastCallTime + interval) - currentTime` and this logic has `interval - (currentTime - lastCallTime) = interval - currentTime + lastCallTime`